test/e2e/utils/recordset-utils.ts

```diff
@@ -390,3 +401,477 @@ export async function testFacetOptionsAndModalRows(page: Page, facetIdx: number,
     await expect.soft(modal).not.toBeAttached();
   });
 }
+
+type TestIndividualFacetParamsGeneral = {
+  /**
+   * the index of facet in the reference.facetColumns list
+   */
+  index: number,
+  /**
+   * the name of the facet
+   */
+  name: string,
+  /**
+   * number of rows on the page after selecting the not-null option
+   */
+  notNullNumRows?: number,
+  /**
+   * number of rows on the page after selecting the null option
+   */
+  nullNumRows?: number,
+};
+
+type TestIndividualFacetParamsCheckPresence = TestIndividualFacetParamsGeneral & {
+  type: 'check_presence',
+  /**
+   * the filter that will be displayed once the not-null option is selected
+   */
+  notNullFilter: string,
+  /**
+   * the filter that will be displayed once the null option is selected
+   */
+  nullFilter: string
+}
+
+type TestIndividualFacetParamsChoice = TestIndividualFacetParamsGeneral & {
+  type: 'choice',
+  isBoolean?: boolean,
+  isEntityMode?: boolean,
+  searchPlaceholder?: string,
+  /**
+   * initially displayed option
+   */
+  options: string[],
+  /**
+   * the option that should be selected
+   */
+  option: number,
+  /**
+   * the filter that will be displayed once the option is selected
+   */
+  filter: string,
+  /**
+   * number of rows on the page after selecting the option
+   */
+  numRows: number,
+}
+
+type TestIndividualFacetParamsRange = TestIndividualFacetParamsGeneral & {
+  type: 'numeric' | 'date' | 'timestamp',
+
+  /**
+   * number of already existing options.
+   */
+  listElems: number,
+  /**
+   * initial value in the min input
+   */
+  initialMin: string | { date: string, time: string },
+  /**
+   * initial value in the max input
+   */
+  initialMax: string | { date: string, time: string },
+  /**
+   * an invalid value to test the error handling.
+   */
+  invalid?: string,
+  /**
+   * The error that will be displayed if the input is invalid
+   */
+  error: string,
+
+  range: {
+    min: string | { date: string, time: string },
+    max: string | { date: string, time: string },
+    /**
+     * the filter that will be displayed after submitting this range
+     */
+    filter: string,
+    /**
+     * number of rows that will be displayed after submitting this range.
+     */
+    numRows: number
+  },
+
+  justMin: {
+    min: string | { date: string, time: string },
+    filter: string,
+    numRows: number
+  },
+
+  justMax: {
+    max: string | { date: string, time: string },
+    filter: string,
+    numRows: number
+  },
+
+}
+
+export type TestIndividualFacetParams = Either<
```

Comment:
Heh, another use of Either except now with 3 types to consider. So this looks like:
```
Either<A, Either<B, C>>
```

Reply:
Yeah exactly. This actually caused a bit of issue for me as this branch was old and I had a bunch of uncommitted changes. I had to ignore my local uncommitted changes and merge with master first.